lib/evaluate.js

```diff
@@ -279,6 +296,108 @@ const calculateInetGroupSuccessRates = (taskGroups) => {
   return result
 }
 
+/**
+ * @param {import('./typings').Measurement[]} measurements
+ * @param {import('./typings').Point} telemetryPoint
+ */
+const reportRetrievalStats = (measurements, telemetryPoint) => {
+  const totalCount = measurements.length
+  if (totalCount < 1) {
+    telemetryPoint.intField('measurements', totalCount)
+    telemetryPoint.intField('unique_tasks', 0)
+    return
+  }
+
+  const uniqueTasksCount = countUniqueTasks(measurements)
+
+  // Calculate aggregates per retrieval result
+  /** @type {Record<import('./typings').RetrievalResult, number> */
+  const resultBreakdown = {
+    OK: 0,
+    TIMEOUT: 0,
+    CAR_TOO_LARGE: 0,
+    BAD_GATEWAY: 0,
+    GATEWAY_TIMEOUT: 0
+  }
+
+  const participants = new Set()
+  const inetGroups = new Set()
+  let downloadBandwidth = 0
+
+  const ttfbHistogram = createHistogram()
+  const durationHistogram = createHistogram()
+  const sizeHistogram = createHistogram()
+
+  for (const m of measurements) {
+    const result = m.retrievalResult ?? 'UNDEFINED'
+    const oldCount = resultBreakdown[result] ?? 0
+    resultBreakdown[result] = oldCount + 1
+
+    participants.add(m.participantAddress)
+    inetGroups.add(m.inet_group)
+
+    // don't trust the checker to submit a positive integers
+    // TODO: reject measurements with invalid values during the preprocess phase?
+    const size = typeof m.byte_length === 'number' && m.byte_length >= 0 ? m.byte_length : undefined
```

Comment:
can you please add () or indentation to make the logic tree more clear?

Reply:
also, what do you think about it `byteLength` instead of `size`, to be consistent?